Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -734,3 +763,165 @@ getStats driverId date = do
+      Just a -> return a
+    unless (authValueHash == req.otp) $ throwError InvalidAuthData
+    value <- Redis.get (makeAlternatePhoneNumberKey personId)
+    altMobNo <- case value of
+      Nothing -> throwError $ InternalError "Alternate Number not found"
+      Just a -> return a
+    encNewNum <- encrypt altMobNo
+    let driver =
+          person
+            { SP.unencryptedAlternateMobileNumber = Just altMobNo,
+              SP.alternateMobileNumber = Just encNewNum
+            }
+    Esq.runTransaction $ do
+      QPerson.updateAlternateMobileNumberAndCode driver
+  return Success
+
+resendOtp ::
+  ( HasFlowEnv m r ["apiRateLimitOptions" ::: APIRateLimitOptions, "smsCfg" ::: SmsConfig],
+    HasFlowEnv m r '["altenateOtpSmsTemplate" ::: Text],
+    EsqDBFlow m r,
+    EncFlow m r,
+    CacheFlow m r,
+    CoreMetrics m,
+    Redis.HedisFlow m r
+  ) =>
+  Id SP.Person ->
+  DriverAlternateNumberReq ->
+  m ResendAuth
+resendOtp personId req = do
+  attempts :: Int <- do
+    res <- Redis.get (makeAlternateNumberAttemptsKey personId)
+    return $ fromMaybe 0 res
+  logDebug $ "attempts" <> show attempts
+  unless (attempts > 0) $ throwError $ AuthBlocked "Attempts limit exceed."
+  smsCfg <- asks (.smsCfg)
+  otpSmsTemplate <- asks (.altenateOtpSmsTemplate)
+  person <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  let altNumber = req.alternateNumber
+      counCode = req.mobileCountryCode
+  otpCode <-
+    Redis.get (makeAlternateNumberOtpKey personId) >>= \case
+      Nothing -> do
+        let useFakeOtpM = useFakeSms smsCfg
+        let fakeOtp = show <$> useFakeOtpM
+        newOtp <- maybe generateOTPCode return fakeOtp
+        expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+        Redis.setExp (makeAlternateNumberOtpKey personId) newOtp expTime
+        return newOtp
+      Just a -> return a
+  let otpHash = smsCfg.credConfig.otpHash
+      altphoneNumber = counCode <> altNumber
+      sender = smsCfg.sender
+  withLogTag ("personId_" <> getId personId) $
+    Sms.sendSMS person.merchantId (Sms.constructSendSMSReq otpCode otpHash otpSmsTemplate altphoneNumber sender)
+      >>= Sms.checkSmsResult
+  updAttempts <- Redis.decrby (makeAlternateNumberAttemptsKey personId) 1
+  let updAttempt = fromIntegral updAttempts
+  return $ ResendAuth {auth = otpCode, attempts = updAttempt}
+
+removeAlternateNumber ::
+  ( EsqDBFlow m r,
+    EncFlow m r,
+    Redis.HedisFlow m r,
+    HasCacheConfig r,
+    CoreMetrics m,
+    HasAternateNumberAttempts r
+  ) =>
+  Id SP.Person ->
+  m APISuccess
+removeAlternateNumber personId = do
+  person <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  let driver =
+        person
+          { SP.unencryptedAlternateMobileNumber = Nothing,
+            SP.alternateMobileNumber = Nothing
+          }
+  Esq.runTransaction $ do
+    QPerson.updateAlternateMobileNumberAndCode driver
+  return Success
```

Comment:
Personally I don't like such huge modules. Maybe we should make some decomposition, like create separate `Domain.Action.UI.Driver.AlternateNumber` module?

On API level we also can put:

```
   :<|> "alternateNumber"
     :> ( "verify"
           :> TokenAuth
           :> ReqBody '[JSON] DDriver.DriverAlternateNumberReq
           :> Post '[JSON] APISuccess
             :<|> "validate"
               :> TokenAuth....
```

But i don't know, which is API requirements